xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/pvr/Recordings.h

```diff
@@ -0,0 +1,473 @@
+  /// conform genre types or something special.
+  ///
+  /// --------------------------------------------------------------------------
+  ///
+  /// **Example:**
+  /// ~~~~~~~~~~~~~{.cpp}
+  /// kodi::addon::PVRRecording tag;
+  /// tag.SetGenreType(EPG_GENRE_USE_STRING);
+  /// tag.SetGenreDescription("Action" + EPG_STRING_TOKEN_SEPARATOR + "Thriller");
+  /// ~~~~~~~~~~~~~
+  ///
+  void SetGenreDescription(const std::string& genreDescription)
+  {
+    strncpy(m_cStructure->strGenreDescription, genreDescription.c_str(),
+            sizeof(m_cStructure->strGenreDescription) - 1);
+  }
+
+  /// @brief To get with @ref SetGenreDescription changed values.
+  std::string GetGenreDescription() const { return m_cStructure->strGenreDescription; }
+
+  /// @brief **optional**\n
+  /// Play count of this recording on the client.
+  void SetPlayCount(int playCount) { m_cStructure->iPlayCount = playCount; }
+
+  /// @brief To get with @ref SetPlayCount changed values.
+  int GetPlayCount() const { return m_cStructure->iPlayCount; }
+
+  /// @brief **optional**\n
+  /// Last played position of this recording on the client.
+  void SetLastPlayedPosition(int lastPlayedPosition)
+  {
+    m_cStructure->iLastPlayedPosition = lastPlayedPosition;
+  }
+
+  /// @brief To get with @ref SetLastPlayedPosition changed values.
+  int GetLastPlayedPosition() const { return m_cStructure->iLastPlayedPosition; }
+
+  /// @brief **optional**\n
+  /// Shows this recording is deleted and can be undelete.
+  void SetIsDeleted(int isDeleted) { m_cStructure->bIsDeleted = isDeleted; }
+
+  /// @brief To get with @ref SetIsDeleted changed values.
+  int GetIsDeleted() const { return m_cStructure->bIsDeleted; }
+
+  /// @brief **optional**\n
+  /// EPG event id associated with this recording. Valid ids must be greater than @ref EPG_TAG_INVALID_UID.
+  void SetEPGEventId(unsigned int epgEventId) { m_cStructure->iEpgEventId = epgEventId; }
+
+  /// @brief To get with @ref SetEPGEventId changed values.
+  unsigned int GetEPGEventId() const { return m_cStructure->iEpgEventId; }
+
+  /// @brief **optional**\n
+  /// Unique identifier of the channel for this recording. @ref PVR_CHANNEL_INVALID_UID
+  /// denotes that channel uid is not available.
+  void SetChannelUid(int channelUid) { m_cStructure->iChannelUid = channelUid; }
+
+  /// @brief To get with @ref SetChannelUid changed values
+  int GetChannelUid() const { return m_cStructure->iChannelUid; }
+
+  /// @brief **optional**\n
+  /// Channel type.
+  ///
+  /// Set to @ref PVR_RECORDING_CHANNEL_TYPE_UNKNOWN if the type cannot be
+  /// determined.
+  ///
+  /// --------------------------------------------------------------------------
+  ///
+  /// Example:
+  /// ~~~~~~~~~~~~~{.cpp}
+  /// kodi::addon::PVRRecording tag;
+  /// tag.SetChannelType(PVR_RECORDING_CHANNEL_TYPE_TV);
+  /// ~~~~~~~~~~~~~
+  ///
+  void SetChannelType(PVR_RECORDING_CHANNEL_TYPE channelType)
```

Comment:
Here, you break encapsulation as you expose a C API data type. This looks wrong.